plugins/modules/ec2_eni_info.py

```diff
@@ -205,27 +205,35 @@
 from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
 
 
-def list_eni(connection, module):
+def build_request_args(eni_id, filters):
+    request_args = {
+        'NetworkInterfaceIds': [eni_id] if eni_id else '',
```

Comment:
I'm surprise you default on an empty string if `eni_id` is False. I would expect an empty list.